tests/LinAlg/matrixTests.hpp

```diff
@@ -71,32 +73,60 @@ class MatrixTests : public TestBase
   MatrixTests() {}
   virtual ~MatrixTests(){}
 
-  int matrixSetToZero(hiop::hiopMatrix& A, const int rank=0)
+  int matrixSetToZero(hiop::hiopMatrixDense& A, const int rank)
```

Comment:
It seems that matrixTests are only used for dense matrix now, while its original design may want to support both dense or sparse input.
I can see we have 2 branches of classes:
1. TestBase->MatrixTests->MatrixTestsDense or MatrixTestsRAJADense 
2. TestBase->MatrixSparseTests->MatrixTestsSparseTriplet or MatrixTestsRAJASparseTriplet 
It is a little bit confusing and hard to tell from the name that MatrixTestsDense inherits from MatrixTests, while MatrixSparseTests doesn't.
To clarify and to keep consistency, how about change the 1st branch to something like:
TestBase->MatrixTestsDense->MatrixTestsDenseRowMajor or MatrixTestsRAJADense

Reply:
This is awkward naming scheme. `MatrixTests` refer to dense matrix tests. There is no base class for both dense and sparse matrix tests because the two sets of tests are so different that it does not make sense to have a base class. 

I think now is probably the best time to fix the confusing naming scheme.

Reply:
Would you like our team to resolve this internally concurrent to this PR or should we incorporate into this PR?

Reply:
concurrently, as I would like to merge this PR asap